packages/trader/src/Modules/Trading/Components/Form/Purchase/__tests__/cancel-deal-info.spec.tsx

```diff
@@ -0,0 +1,79 @@
+
+const deal_cancellation = 'Deal cancel. fee';
+const price = '1,023.00 USD';
+const mock_proposal_info = {
+    id: '129106862',
+    cancellation: { ask_price: 1023, date_expiry: 1907128726 },
+    has_error: false,
+} as unknown as TProposalTypeInfo;
+
+const default_mock_store = {
+    modules: {
+        trade: {
+            currency: 'USD',
+            has_cancellation: false,
+        },
+    },
+};
+
+describe('<CancelDealInfo />', () => {
+    const mockCancelDealInfo = (mocked_store: TCoreStores, mock_proposal_info: TProposalTypeInfo) => {
+        return (
+            <TraderProviders store={mocked_store}>
+                <CancelDealInfo proposal_info={mock_proposal_info} data_testid='cancellation-wrapper' />
+            </TraderProviders>
+        );
+    };
+
+    it('should not render component if has_cancellation is false', () => {
+        const { container } = render(mockCancelDealInfo(mockStore(default_mock_store), mock_proposal_info));
+
+        expect(container).toBeEmptyDOMElement();
+    });
+    it('should render component if has_cancellation is true', () => {
+        default_mock_store.modules.trade.has_cancellation = true;
+        render(mockCancelDealInfo(mockStore(default_mock_store), mock_proposal_info));
+
+        expect(screen.getByText(deal_cancellation)).toBeInTheDocument();
+        expect(screen.getByText(price)).toBeInTheDocument();
+    });
+    it('should not render Money component if has_error is true', () => {
+        mock_proposal_info.has_error = true;
+        render(mockCancelDealInfo(mockStore(default_mock_store), mock_proposal_info));
+
+        expect(screen.getByText(deal_cancellation)).toBeInTheDocument();
+        expect(screen.queryByText(price)).not.toBeInTheDocument();
+    });
+    it('should not render Money component if id is falsy and has_error is null or undefined', () => {
+        mock_proposal_info.has_error = null as unknown as TProposalTypeInfo['has_error'];
+        mock_proposal_info.id = '';
+        render(mockCancelDealInfo(mockStore(default_mock_store), mock_proposal_info));
+
+        expect(screen.getByText(deal_cancellation)).toBeInTheDocument();
+        expect(screen.queryByText(price)).not.toBeInTheDocument();
+    });
+    it('should apply special className if clientHeight > 21', () => {
+        const originalClientHeight = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'clientHeight');
+        Object.defineProperty(HTMLElement.prototype, 'clientHeight', {
+            configurable: true,
+            value: 200,
+        });
+        render(mockCancelDealInfo(mockStore(default_mock_store), mock_proposal_info));
+
+        expect(screen.getByTestId('cancellation-wrapper')).toHaveClass(
+            'trade-container__cancel-deal-info trade-container__cancel-deal-info--row-layout'
+        );
+
+        Object.defineProperty(HTMLElement.prototype, 'clientHeight', {
+            configurable: true,
```

Comment:
@kate-deriv since it's only needed in 1 test case, it's ok to clean it up inside of it. However, I think no need to add `configurable: true` here the 2nd time, according to the example of packages/account/src/Sections/Security/ConnectedApps/__tests__/connected-apps.spec.js :)